gemonlinedb/include/gem/onlinedb/SystemTopology.h

```diff
@@ -0,0 +1,136 @@
+#ifndef GEM_ONLINEDB_SYSTEMTOPOLOGY_H
+#define GEM_ONLINEDB_SYSTEMTOPOLOGY_H
+
+#include <array>
+#include <vector>
+
+#include "gem/onlinedb/AMC13Configuration.h"
+#include "gem/onlinedb/AMCConfiguration.h"
+#include "gem/onlinedb/OHv3Configuration.h"
+#include "gem/onlinedb/VFAT3ChipConfiguration.h"
+
+#include "gem/onlinedb/ConfigurationTraits.h"
+#include "gem/onlinedb/PartReference.h"
+
+namespace gem {
+    namespace onlinedb {
+
+        /**
+         * @brief Describes the topology of a system.
+         *
+         * The description is built as a tree of nodes containing part
+         * references for the various components. The root node is the AMC 13
+         * (only one AMC 13 is supported), under which AMCs are found, then OHs
+         * and VFATs.
+         *
+         * The tree can be read from an XML document that adheres to the
+         * @c system-topology schema.
+         */
+        class SystemTopology
+        {
+        public:
+            /**
+             * @brief Represents a VFAT 3.
+             */
+            class VFAT3Node
+            {
+                friend class SystemTopology;
+
+            public:
+                /**
+                 * @brief The part reference associated to the VFAT.
+                 * @note The @c PartType for @ref VFAT3ChannelConfiguration
+                 *       isn't used.
+                 */
+                ConfigurationTraits<VFAT3ChipConfiguration>::PartType reference;
+
+            private:
+                /// @brief Populates a @c VFAT3Node from XML data.
+                void populate(const xercesc::DOMElement *el);
+            };
+
+            /**
+             * @brief Represents an OHv3.
+             */
+            class OHv3Node
```

Comment:
I would try to keep the names as generic as possible. E.g. this code should be reusable by e.g. GE2/1 which is quite similar to GE1/1, but will have different OH and different number of VFATs

Reply:
We can decide to rename it to "OHNode" but I'd do this in a separate PR since there are quite a few things named OHv3* already. Also non-trivial design decisions would need to be made if two separate OH classes are needed.

I fear that supporting both GE2/1 and GE1/1 in the same build would require some heavy refactoring.

Reply:
the firmware will hide the hardware changes at max, so I doubt there will be non-trivial design decisions required.

Reply:
I feared for a moment that I'd need to refactor the 2,000 lines already committed to this branch... I still think that the naming convention shouldn't be changed in this PR.